src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsWriter.cs

```diff
@@ -0,0 +1,249 @@
+            => WriteIntAsync(stream, (int)value, isAsync, ct);
+
+        /// <summary>
+        /// Writes long value to out buffer, as little-endian.
+        /// </summary>
+        /// <param name="stream">Tds Stream to work with</param>
+        /// <param name="value">Value to write</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public async ValueTask WriteLongAsync(TdsStream stream, long value, bool isAsync, CancellationToken ct)
+        {
+            var len = sizeof(long);
+            BinaryPrimitives.WriteInt64LittleEndian(_alloc.GetBuffer(len), value);
+            await WriteBytesAsync(stream, _alloc.GetBuffer(len), 0, len, isAsync, ct).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Writes unsigned long value to out buffer, as little-endian.
+        /// </summary>
+        /// <param name="stream">Tds Stream to work with</param>
+        /// <param name="value">Value to write</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public ValueTask WriteUnsignedLongAsync(TdsStream stream, ulong value, bool isAsync, CancellationToken ct)
+            => WriteLongAsync(stream, (long)value, isAsync, ct);
+
+        /// <summary>
+        /// Writes float value to out buffer, as little-endian.
+        /// </summary>
+        /// <param name="stream">Tds Stream to work with</param>
+        /// <param name="value">Value to write</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public async ValueTask WriteFloatAsync(TdsStream stream, float value, bool isAsync, CancellationToken ct)
+        {
+            Debug.Assert(!float.IsInfinity(value) && !float.IsNaN(value), "Float value is out of range.");
+
+            var len = sizeof(float);
+            BinaryPrimitives.WriteInt32LittleEndian(_alloc.GetBuffer(len), BitConverter.SingleToInt32Bits(value));
+            await WriteBytesAsync(stream, _alloc.GetBuffer(len), 0, len, isAsync, ct).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Writes double value to out buffer, as little-endian.
+        /// </summary>
+        /// <param name="stream">Tds Stream to work with</param>
+        /// <param name="value">Value to write</param>
+        /// <param name="isAsync">Whether caller method is executing asynchronously.</param>
+        /// <param name="ct">Cancellation token</param>
+        public async ValueTask WriteDoubleAsync(TdsStream stream, double value, bool isAsync, CancellationToken ct)
+        {
+            Debug.Assert(!double.IsInfinity(value) && !double.IsNaN(value), "Double value is out of range.");
```

Comment:
`!double.IsInfinity(value) && !double.IsNaN(value)` can be replaced with `double.IsFinite(value)`.